clients/src/main/java/org/apache/kafka/clients/consumer/internals/CompletedFetch.java

```diff
@@ -311,25 +311,35 @@ <K, V> ConsumerRecord<K, V> parseRecord(Deserializers<K, V> deserializers,
                                             Optional<Integer> leaderEpoch,
                                             TimestampType timestampType,
                                             Record record) {
+        long offset = record.offset();
+        long timestamp = record.timestamp();
+        ByteBuffer keyBytes = record.key();
+        ByteBuffer valueBytes = record.value();
+        Headers headers = new RecordHeaders(record.headers());
+        K key;
+        V value;
         try {
-            long offset = record.offset();
-            long timestamp = record.timestamp();
-            Headers headers = new RecordHeaders(record.headers());
-            ByteBuffer keyBytes = record.key();
-            K key = keyBytes == null ? null : deserializers.keyDeserializer.deserialize(partition.topic(), headers, keyBytes);
-            ByteBuffer valueBytes = record.value();
-            V value = valueBytes == null ? null : deserializers.valueDeserializer.deserialize(partition.topic(), headers, valueBytes);
-            return new ConsumerRecord<>(partition.topic(), partition.partition(), offset,
-                    timestamp, timestampType,
-                    keyBytes == null ? ConsumerRecord.NULL_SIZE : keyBytes.remaining(),
-                    valueBytes == null ? ConsumerRecord.NULL_SIZE : valueBytes.remaining(),
-                    key, value, headers, leaderEpoch);
+            key = keyBytes == null ? null : deserializers.keyDeserializer.deserialize(partition.topic(), headers, keyBytes);
         } catch (RuntimeException e) {
-            log.error("Deserializers with error: {}", deserializers);
-            throw new RecordDeserializationException(partition, record.offset(),
-                    "Error deserializing key/value for partition " + partition +
+            throw new RecordDeserializationException(RecordDeserializationException.DeserializationExceptionOrigin.KEY,
```

Comment:
What about using a helper method like this to throw the exception?

```sh
    private void throwRecordDeserializationException(RecordDeserializationException.DeserializationExceptionOrigin origin,
                                                     TopicPartition partition,
                                                     TimestampType timestampType,
                                                     Record record,
                                                     RuntimeException e) {
        throw new RecordDeserializationException(origin, partition, record.offset(), record.timestamp(), timestampType, record.key(), 
            record.value(), new RecordHeaders(record.headers()), "Error deserializing " + origin.toString() + " for partition " 
                + partition + " at offset " + record.offset() + ". If needed, please seek past the record to continue consumption.", e);
    }
```

Reply:
Done but with method to create the exception as static code analysis was a bit disappointed otherwise.